arelle/plugin/validate/EDINET/rules/upload.py

```diff
@@ -54,3 +56,96 @@ def rule_EC0121E(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[DISCLOSURE_SYSTEM_EDINET],
+)
+def rule_EC0124E(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    EDINET.EC0124E: There are no empty directories.
+
+    See note on EC0121E.
+    """
+    if not pluginData.shouldValidateUpload(val):
+        return
+    uploadFilepaths = pluginData.getUploadFilepaths(val.modelXbrl)
+    emptyDirectories = []
+    for path in uploadFilepaths:
+        if path.suffix:
+            continue
+        if not any(path in p.parents for p in uploadFilepaths):
+            emptyDirectories.append(str(path))
+    for emptyDirectory in emptyDirectories:
+        yield Validation.error(
+            codes='EDINET.EC0124E',
+            msg=_("There is no file directly under '%(emptyDirectory)s'. "
+                  "No empty folders. "
+                  "Please store the file in the appropriate folder or delete the folder and upload again."),
+            emptyDirectory=emptyDirectory,
+        )
+
+
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[DISCLOSURE_SYSTEM_EDINET],
+)
+def rule_EC0132E(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    EDINET.EC0132E: Store the manifest file directly under the relevant folder.
+    """
+    if not pluginData.shouldValidateUpload(val):
+        return
+    uploadFilepaths = pluginData.getUploadFilepaths(val.modelXbrl)
+    docFolders = ("PublicDoc", "PrivateDoc", "AuditDoc")
+    for filepath in uploadFilepaths:
+        if filepath.name not in docFolders:
+            continue
+        expectedManifestName = f'manifest_{filepath.name}.xml'
+        expectedManifestPath = filepath / expectedManifestName
+        if expectedManifestPath in uploadFilepaths:
+            continue
+        yield Validation.error(
+            codes='EDINET.EC0132E',
+            msg=_("'%(expectedManifestName)s' does not exist in '%(expectedManifestFolder)s'. "
+                  "Please store the manifest file (or cover file) directly under the relevant folder and upload it again. "),
+            expectedManifestName=expectedManifestName,
+            expectedManifestFolder=str(filepath),
+        )
+
+
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[DISCLOSURE_SYSTEM_EDINET],
+)
+def rule_EC0183E(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    EDINET.EC0183E: The compressed file size exceeds 55MB.
+    """
+    if not pluginData.shouldValidateUpload(val):
+        return
+    zipFile = val.modelXbrl.fileSource.fs
+    zipFile.fp.seek(0, 2)  # Move to the end of the file
+    size = zipFile.fp.tell()
+    if size > 55 * 1024 * 1024:
```

Comment:
Can this same `seek` technique be used to improve [this logic](https://github.com/Arelle/Arelle/pull/1689#discussion_r2113786796)?  Perhaps a helper should be added to `FileSource`.

Similar to [here](https://github.com/Arelle/Arelle/pull/1689#discussion_r2113768670), is the limit MB or MiB?  The code is checking the latter.